src/extension_commands.ml

```diff
@@ -96,6 +96,89 @@ let _upgrade_ocaml_lsp_server =
   command Extension_consts.Commands.upgrade_ocaml_lsp_server handler
 ;;
 
+let _install_dune_lsp_server =
+  let handler (instance : Extension_instance.t) ~args:_ =
+    let open Promise.Syntax in
+    let _ =
+      match Extension_instance.sandbox instance with
+      | Dune dune ->
+        let* is_dune_locked = Dune.detect_dune_lock_dir (Dune.root dune) () in
+        if is_dune_locked
+        then
+          let* ocamllsp_present = Extension_instance.check_ocaml_lsp_available instance in
```

Comment:
Here, inside your `install_dune_lsp` command, you're calling `check_ocaml_lsp_available`.